src/app/(proper_react)/(onboarding)/user/welcome/EnterInfo.tsx

```diff
@@ -4,22 +4,357 @@
+                />
+              ),
+              privacy_protection_link: (
+                <a
+                  href="https://www.mozilla.org/firefox/privacy"
+                  target="_blank"
+                />
+              ),
+            },
+          })}
+        </p>
+
+        <div className={viewStyles.confirmButtonWrapper}>
+          <Button
+            type="primary"
+            onClick={() => explainerDialogState.close()}
+            autoFocus={true}
+            className={viewStyles.startButton}
+          >
+            {l10n.getString(
+              "onboarding-get-started-how-it-works-dialog-confirm-label"
+            )}
+          </Button>
+        </div>
+      </div>
+    </Dialog>
+  );
+
+  // TODO: Move dialogs into its own component
+  const ConfirmInfoDialog = () => (
+    <Dialog
+      title={l10n.getString("onboarding-enter-details-comfirm-dialog-title")}
+    >
+      <p>{l10n.getString("onboarding-enter-details-comfirm-dialog-text")}</p>
+      <div className={viewStyles.dialogContents}>
+        <ul className={enterInfoStyles.infoList}>
+          {userDetailsData.map(({ key, label, displayValue }) => (
+            <li key={key} className={enterInfoStyles.infoItem}>
+              {label}: <strong>{displayValue}</strong>
+            </li>
+          ))}
+        </ul>
+      </div>
+      <div className={viewStyles.stepButtonWrapper}>
+        <Button
+          type="secondary"
+          onClick={() => confirmDialogState.close()}
+          className={enterInfoStyles.startButton}
+        >
+          {l10n.getString(
+            "onboarding-enter-details-comfirm-dialog-button-edit"
+          )}
+        </Button>
+        <Button
+          type="primary"
+          onClick={() => props.onDataSaved()}
+          autoFocus={true}
+          className={enterInfoStyles.startButton}
+        >
+          {l10n.getString(
+            "onboarding-enter-details-comfirm-dialog-button-confirm"
+          )}
+        </Button>
+      </div>
+    </Dialog>
+  );
 
   return (
-    <div className={styles.wrapper}>
-      Enter your info here. After it&apos;s sent to the backend, we&apos;ll call{" "}
-      <code>props.onDataSaved</code>.
+    <div className={viewStyles.stepContent}>
+      <h1>{l10n.getString("onboarding-enter-details-title")}</h1>
+      <p>
+        {l10n.getString("onboarding-enter-details-text")}
+        <button
+          {...explainerDialogTrigger.triggerProps}
+          onClick={() => explainerDialogState.open()}
+          className={viewStyles.explainerTrigger}
+        >
+          {l10n.getString("onboarding-enter-details-why-button-label")}
+        </button>
+      </p>
+      <div className={enterInfoStyles.inputContainer}>
+        {userDetailsData.map(
+          ({
+            key,
+            errorMessage,
+            label,
+            onChange,
+            placeholder,
+            type,
+            value,
+          }) => (
+            <TextField
```

Comment:
A dedicated input for the location data autocompletion will be added after https://github.com/mozilla/blurts-server/pull/3186 landed.